osu.Game/Online/Leaderboards/Leaderboard.cs

```diff
@@ -39,7 +39,9 @@ public abstract class Leaderboard<TScope, TScoreInfo> : CompositeDrawable
         /// <summary>
         /// The currently displayed scores.
         /// </summary>
-        public IEnumerable<TScoreInfo> Scores => scores;
+        public IBindableList<TScoreInfo> Scores => scores;
+
+        private readonly BindableList<TScoreInfo> scores = new BindableList<TScoreInfo>();
```

Comment:
This doesn't seem to reset when selecting another beatmap while on "Details" section, causing the gameplay session to show a leaderboard with scores from another beatmap.

Reply:
I've fixed this by clearing the old scores for now.

I think in the future we'd either want the leaderboard to be always present at song select (ie. with the new design), or the retrieval should be done by its own component (as @smoogipoo touched on earlier).